### PR TITLE
feat(service-client-sample): adding sample service with client

### PR DESCRIPTION
**Pull request by whoswendy**
This PR contains code to add a new sample reactive service integrated with the reactive client weather module to demo integration of service and client modules.

### CODE REVIEW

File: service/service-samples/pom.xml [hunk 1]
```diff
@@ -24,6 +24,7 @@
         <module>sample-service-rest-mysql-book</module>
         <module>sample-service-rest-oracle-book</module>
         <module>sample-service-rest-postgres-book</module>
+        <module>sample-service-reactive-weather-client</module>
```
Comment (andCastStatic) [inline service/service-samples/pom.xml]: May this be in alphabetical order?

File: client/client-samples/sample-client-reactive-weather/src/main/java/io/americanexpress/sample/client/reactive/weather/model/CurrentWeather.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    /**
+     * Sets time.
+     *
+     * @param time the time
+     */
+    public void setTime(LocalDateTime time) {
+        this.time = time;
+    }
+
+    private LocalDateTime time;
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-weather/src/main/java/io/americanexpress/sample/client/reactive/weather/model/CurrentWeather.java]: Could you move this variable up and add javadoc

File: service/service-samples/sample-service-reactive-weather-client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            <artifactId>synapse-service-rest</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.americanexpress.synapse</groupId>
+            <artifactId>sample-client-reactive-weather</artifactId>
+            <version>0.3.8-SNAPSHOT</version>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (andCastStatic) [inline service/service-samples/sample-service-reactive-weather-client/pom.xml]: add a new line please.

Comment (guttff) [inline service/service-samples/sample-service-reactive-weather-client/pom.xml]: add extra line

File: service/service-samples/sample-service-reactive-weather-client/src/main/java/io/americanexpress/service/weather/rest/model/GetWeatherResponse.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * the License.
+ */
+package io.americanexpress.service.weather.rest.model;
+
+import io.americanexpress.synapse.service.rest.model.BaseServiceResponse;
+
+/**
+ * {@code GetWeatherResponse} is the response for GET weather.
+ */
+public class GetWeatherResponse extends BaseServiceResponse {
```
Comment (andCastStatic) [inline service/service-samples/sample-service-reactive-weather-client/src/main/java/io/americanexpress/service/weather/rest/model/GetWeatherResponse.java:21]: add missing javadocs please.

File: service/service-samples/sample-service-reactive-weather-client/src/main/java/io/americanexpress/service/weather/rest/service/GetWeatherService.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+     * Instantiates a new Get weather service.
+     *
+     * @param weatherReactiveClient the weather reactive client
+     */
+    public GetWeatherService(WeatherReactiveClient weatherReactiveClient) {
+        this.weatherReactiveClient = weatherReactiveClient;
+    }
+
+    @Override
+    protected Mono<GetWeatherResponse> executeRead(HttpHeaders headers, String request) {
```
Comment (andCastStatic) [inline service/service-samples/sample-service-reactive-weather-client/src/main/java/io/americanexpress/service/weather/rest/service/GetWeatherService.java:53]: add missing javadocs please.

File: service/service-samples/sample-service-reactive-weather-client/src/main/java/io/americanexpress/service/weather/rest/service/GetWeatherService.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+/**
+ * {@code GetWeatherService} is the service for getting the current temperature and weather.
+ */
+@Service
+public class GetWeatherService extends BaseGetMonoReactiveService<GetWeatherResponse> {
+
+    private final WeatherReactiveClient weatherReactiveClient;
```
Comment (andCastStatic) [inline service/service-samples/sample-service-reactive-weather-client/src/main/java/io/americanexpress/service/weather/rest/service/GetWeatherService.java:35]: add missing javadocs please.

File: client/client-samples/sample-client-reactive-weather/src/main/java/io/americanexpress/sample/client/reactive/weather/model/CurrentWeather.java (hunk 2 above)
Comment (whoswendy) [reply client/client-samples/sample-client-reactive-weather/src/main/java/io/americanexpress/sample/client/reactive/weather/model/CurrentWeather.java]: done

File: service/service-samples/sample-service-reactive-weather-client/src/main/java/io/americanexpress/service/weather/rest/service/GetWeatherService.java (hunk 6 above)
Comment (guttff) [inline service/service-samples/sample-service-reactive-weather-client/src/main/java/io/americanexpress/service/weather/rest/service/GetWeatherService.java:35]: add javadoc

File: service/service-samples/sample-service-reactive-weather-client/src/main/java/io/americanexpress/service/weather/rest/service/GetWeatherService.java (hunk 5 above)
Comment (guttff) [inline service/service-samples/sample-service-reactive-weather-client/src/main/java/io/americanexpress/service/weather/rest/service/GetWeatherService.java:53]: add javadoc

**Review by guttff — APPROVED** [review]

**Pull request merged**